src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginUpdateHeader.java

```diff
@@ -0,0 +1,107 @@
+package com.github.manolo8.darkbot.gui.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.plugins.PluginHandler;
+import com.github.manolo8.darkbot.extensions.plugins.PluginUpdater;
+import com.github.manolo8.darkbot.gui.components.MainButton;
+import com.github.manolo8.darkbot.utils.I18n;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class PluginUpdateHeader extends JPanel {
+
+    private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("EEE, d MMM, hh:mm a", Locale.ROOT);
+
+    private final PluginUpdater pluginUpdater;
+    private final PluginHandler pluginHandler;
+
+    private final JLabel title;
+    private final JProgressBar progressBar;
+    private final UpdateAllButton updateAllButton;
+    private final CheckUpdateButton checkUpdateButton;
+
+    PluginUpdateHeader(Main main) {
+        super(new MigLayout("ins 0, gap 0, fill", "[grow][][]", "[][grow]"));
+
+        this.pluginHandler = main.pluginHandler;
+        this.pluginUpdater = main.pluginUpdater;
+
+        add(this.title = new JLabel());
+        add(this.updateAllButton = new UpdateAllButton());
+        add(this.checkUpdateButton = new CheckUpdateButton());
+        add(this.progressBar = new JProgressBar(), "dock south, spanx");
+
+        progressBar.setBorderPainted(false);
+
+        refreshUI();
+    }
+
+    void refreshUI() {
+        updateAllButton.refresh();
+        progressBar.setVisible(false);
+
+        checkUpdateButton.setEnabled(true);
+        String status = !pluginUpdater.hasAnyUpdates()
+                ? I18n.get("plugins.config_button.up_to_date")
+                : I18n.get("plugins.config_button.updates_available", pluginHandler.getAvailableUpdates().count());
+        title.setText(I18n.get("plugins.config_button.update_header",
+                status, dateFormatter.format(pluginUpdater.getLastChecked())));
```

Comment:
i'm kinda surprised that the dateFormatter can suck up a `long` directly, but i'd try to use java8 dates for this

Convert the long into an actual instant, then you can use a dateFormatter to format that instant
`Instant lastUpdate = Instant.ofEpochMilli(pluginUpdater.getLastChecked());`

Keep in mind you'll need to swap out the older `SimpleDateFormat` with a java 8 `DateTimeFormatter`.
`DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("EEE, d MMM, hh:mm a");`

Lastly, for i18n purposes you can consider using the date formats provided by java, this way users with different preferences will get the date format in their regional format, example:
`DateTimeFormatter dateFormatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.FULL, FormatStyle.FULL)`

The issue is you're trying to print a date without a year, and that has no real standard (see https://stackoverflow.com/questions/9720750/java-local-timeformat-without-year) so you can probably ignore this for now